### PR TITLE
Update agent manifests based on helm chart 2.27.8

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Update the manifests documented on this page https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset#installation

### Preview

https://docs-staging.datadoghq.com/<BRANCH_NAME>/<PATH>

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by vboulineau — APPROVED** [review]

File: static/resources/yaml/datadog-agent-windows-apm.yaml [hunk 1]
```diff
@@ -211,6 +218,12 @@ spec:
         - hostPath:
             path: \\.\pipe\docker_engine
           name: runtimesocket
+        # if the CRI or is not provived we should try to mount the default containerd pipe.
+        # by default "datadog.dockerOrCriSocketPath" mount the docker pipe.
+        # so with this additional hostPath, by default both are mounted.
```
Comment (alai97) [inline static/resources/yaml/datadog-agent-windows-apm.yaml:223]:
```suggestion
        # If the CRI is not provided, try to mount the default containerd pipe.
        # By default, datadog.dockerOrCriSocketPath mounts the Docker pipe.
        # With this additional hostPath, both are mounted.
```

File: static/resources/yaml/datadog-agent-windows-all-features.yaml [hunk 2]
```diff
@@ -259,18 +270,24 @@ spec:
...
         - hostPath:
-            path: C:/ProgramData/docker/containers
+            path: C:/ProgramData
           name: logdockercontainerpath
+        - hostPath:
+            path: \\.\pipe\docker_engine
+          name: runtimesocket
+        # if the CRI or is not provived we should try to mount the default containerd pipe.
+        # by default "datadog.dockerOrCriSocketPath" mount the docker pipe.
+        # so with this additional hostPath, by default both are mounted.
```
Comment (alai97) [inline static/resources/yaml/datadog-agent-windows-all-features.yaml:287]:
```suggestion
        # If the CRI is not provided, try to mount the default containerd pipe.
        # By default, datadog.dockerOrCriSocketPath mounts the Docker pipe.
        # With this additional hostPath, both are mounted.
```

File: static/resources/yaml/datadog-agent-windows-logs-apm.yaml [hunk 3]
```diff
@@ -222,18 +229,24 @@ spec:
...
         - hostPath:
-            path: C:/ProgramData/docker/containers
+            path: C:/ProgramData
           name: logdockercontainerpath
+        - hostPath:
+            path: \\.\pipe\docker_engine
+          name: runtimesocket
+        # if the CRI or is not provived we should try to mount the default containerd pipe.
+        # by default "datadog.dockerOrCriSocketPath" mount the docker pipe.
+        # so with this additional hostPath, by default both are mounted.
```
Comment (alai97) [inline static/resources/yaml/datadog-agent-windows-logs-apm.yaml:246]:
```suggestion
        # If the CRI is not provided, try to mount the default containerd pipe.
        # By default, datadog.dockerOrCriSocketPath mounts the Docker pipe.
        # With this additional hostPath, both are mounted.
```

File: static/resources/yaml/datadog-agent-windows-logs.yaml [hunk 4]
```diff
@@ -174,18 +179,24 @@ spec:
...
         - hostPath:
-            path: C:/ProgramData/docker/containers
+            path: C:/ProgramData
           name: logdockercontainerpath
+        - hostPath:
+            path: \\.\pipe\docker_engine
+          name: runtimesocket
+        # if the CRI or is not provived we should try to mount the default containerd pipe.
+        # by default "datadog.dockerOrCriSocketPath" mount the docker pipe.
+        # so with this additional hostPath, by default both are mounted.
```
Comment (alai97) [inline static/resources/yaml/datadog-agent-windows-logs.yaml:196]:
```suggestion
        # If the CRI is not provided, try to mount the default containerd pipe.
        # By default, datadog.dockerOrCriSocketPath mounts the Docker pipe.
        # With this additional hostPath, both are mounted.
```

File: static/resources/yaml/datadog-agent-windows-vanilla.yaml [hunk 5]
```diff
@@ -165,6 +170,12 @@ spec:
         - hostPath:
             path: \\.\pipe\docker_engine
           name: runtimesocket
+        # if the CRI or is not provived we should try to mount the default containerd pipe.
+        # by default "datadog.dockerOrCriSocketPath" mount the docker pipe.
+        # so with this additional hostPath, by default both are mounted.
```
Comment (alai97) [inline static/resources/yaml/datadog-agent-windows-vanilla.yaml:175]:
```suggestion
        # If the CRI is not provided, try to mount the default containerd pipe.
        # By default, datadog.dockerOrCriSocketPath mounts the Docker pipe.
        # With this additional hostPath, both are mounted.
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Small comment nit!

**Review by alai97 — APPROVED** [review]

**Pull request merged**